### PR TITLE
Implement support for optionally specifying the 'var' keyword in 'for' loops

### PR DESCRIPTION
**Pull request by gunnarbeutner**
This adds support for _optionally_ using the `var` keyword in for loops:

```
for (var number in [ 4, 8, 15, 16, 23, 42 ]) {
    log(number)
}
```

and

```
for (var key => var value in globals) {
    log(key)
    log(typeof(value))
}
```

Adding the `var` keyword has no semantic effect other than unifying how we usually do variable declarations.

### CODE REVIEW

**Comment by dnsmichi** [comment]
From a first look, this makes it complicated to follow and understand e.g. apply for inside the monitoring basics chapter. Let's talk about this next week in the office :)

**Comment by dnsmichi** [comment]
It can be added, nevertheless we shouldn't change our entire documentation for it. If that's the requirement, I'd drop this feature as this makes it complicated for our users (who are not programmers).

**Comment by gunnarbeutner** [comment]
I guess I _might_ reword this to drop all the docs changes (apart from the bit where we mention that users can optionally add `var` in their loop var decls).

**Comment by gunnarbeutner** [comment]
Any objections to setting the target to 2.10?

**Comment by dnsmichi** [comment]
Looks good, thanks :) Review/merge happens after 2.9.
